user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -595,6 +610,66 @@ def _read_qualification_output_file(self, report_name_key: str, file_format_key:
         report_file_path = FSUtil.build_path(self.ctxt.get_rapids_output_folder(), report_file_name)
         return pd.read_csv(report_file_path)
 
+    def _read_qualification_metric_file(self, file_name: str) -> Dict[str, pd.DataFrame]:
+        """
+        Helper method to read metric files from the qualification tool's output metric folder.
+        Returns a dictionary of DataFrames, where each key is an application ID, and each
+        DataFrame contains the corresponding application's metrics data.
+        Example:
+            {
+                'appId1': pd.DataFrame(...),
+                'appId2': pd.DataFrame(...),
+            }
+       :param file_name: Name of the metric file to read from each application's folder
+       """
+        metrics = {}
+        root_metric_dir = self.ctxt.get_metrics_output_folder()
+        for metric_dir in FSUtil.get_subdirectories(root_metric_dir):
+            app_id_str = FSUtil.get_resource_name(metric_dir)
+            report_file_path = FSUtil.build_path(metric_dir, file_name)
+            try:
+                metrics[app_id_str] = pd.read_csv(report_file_path)
+            except Exception as e:  # pylint: disable=broad-except
+                # Some apps may not have the given metrics file, we should ensure
+                # that the dictionary contains entries for all apps to avoid KeyErrors
+                # and maintain consistency in processing.
+                metrics[app_id_str] = pd.DataFrame()
+                self.logger.warning('Unable to read metrics file for app %s. Reason - %s:%s',
```

Comment:
nit: at some point we need to find a better way to log those warning messages. When I run the tools, there is a ton of messages triggered by missing some of the profiler's output.
We could do improve that in a separate issue. We can buffer all those missing files and just dump a single warning message.

Reply:
Yes, that is true. We could cleanup the logs by buffering the logs instead of displaying per app.

For this PR, buffered the logs by creating a list of apps with missing metrics and displaying a single log message with all app ids.